### DISCUSSION TITLE
Tasmota 12.2.0 which Timestamp for InlfuxDB can i use ?

### DISCUSSION THREAD

**Discussion by Starfoxfs** [arendst/Tasmota#17087] [Q&A] [answered]
Hi,
i use on my Shelly EM actually Tasmota in the latest Version 12.2.0

I write these Data via MQTT and NodeRed into a InfluxDB Ver. 1.8.

Which Timestamp Format from Tasmota can i use to write the values to InfluxDB ?

The Tasmota Commands says this:

`Time | 0 = enable NTP (default)
1 = format JSON message timestamp in ISO format
2 = format JSON message timestamp in both ISO and Epoch format
3 = format JSON message timestamp in Epoch format
4 = format JSON message timestamp in milliseconds
<value> = disable NTP and set UTC time as epoch value if greater than 1451602800 (January 1, 2016)
`

The InfluxDB Documentation says this:

> [Timestamp](https://docs.influxdata.com/influxdb/v1.8/write_protocols/line_protocol_tutorial/#timestamp)

The [timestamp](https://docs.influxdata.com/influxdb/v1.8/concepts/glossary/#timestamp) for your data point in nanosecond-precision Unix time. The timestamp is optional in line protocol. If you do not specify a timestamp for your data point InfluxDB uses the server’s local nanosecond timestamp in UTC.

In the example, the timestamp is 1465839830100400200 (that’s 2016-06-13T17:43:50.1004002Z in RFC3339 format). The line protocol below is the same data point but without the timestamp. When InfluxDB writes it to the database it uses your server’s local timestamp instead of 2016-06-13T17:43:50.1004002Z.

weather,location=us-midwest temperature=82

Use the InfluxDB API to specify timestamps with a precision other than nanoseconds, such as microseconds, milliseconds, or seconds. We recommend using the coarsest precision possible as this can result in significant improvements in compression. See the [API Reference](https://docs.influxdata.com/influxdb/v1.8/tools/api/#write-http-endpoint) for more information.
[Setup Tip:](https://docs.influxdata.com/influxdb/v1.8/write_protocols/line_protocol_tutorial/#setup-tip)

Use the Network Time Protocol (NTP) to synchronize time between hosts. InfluxDB uses a host’s local time in UTC to assign timestamps to data; if hosts’ clocks aren’t synchronized with NTP, the timestamps on the data written to InfluxDB can be inaccurate.

**Comment by joba-1** [comment, marked as answer]
I would not use any timestamp. Then the influx db servers time is used. Works very well for me. I don't have to worry about time (dst, zone, ntp, ) on tasmota devices, just on one server.
Or you use the tasmota integrated functionality (self compiled) where sensor data is formatted for you - also no worries about time format

**Reply by joba-1** [reply]
https://github.com/joba-1/Tasmogy

this python script I use to read energy values of a tasmota nous/gosound plug

Maybe it helps as a starting point (or even as is)

**Comment by Starfoxfs** [comment]
Ok thanks, i tested the Tasmota given Timestamps don´t work to write in InfluxDB.

Now i let write the Servers Timestamp to Influx, works great
